### PR TITLE
All tasks from HW 7

### PR DESCRIPTION
**Pull request by albina-pasichnyk**
In scope of this pull request following tasks were resolved:
Task 7.1 - Custom print function, with ability to print to file.
Task 7.2 - Custom filter function + Tests.
Task 7.3 - Custom map function + Tests.
Task 7.4 - Custom min and custom max functions + Tests.
Task 7.5 - Custom all funtion + Tests.
Task 7.6 - Function for counting arrows rotations + Tests.

### CODE REVIEW

File: HW_7/HW_7.1_Custom_print_function.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        # Open the file in write mode
+        with open(output_file, "w") as file:
+            # Redirect standard output to the file
+            sys.stdout = file
+            # Use sys.stdout.write() to write text to the file
+            for each in args:
+                sys.stdout.write(f'{each}{sep}')
+            sys.stdout.write(end)
+            # Reset the standard output back to the console
+            sys.stdout = sys.__stdout__
```
Comment (bohdandovbsyh) [inline HW_7/HW_7.1_Custom_print_function.py:32]: wow) it is even better than I want from you))

File: HW_7/HW_7.2_Custom_filter_function.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+def custom_filter(callback, sequence):
+    """
+    Function custom_filter is used to filter elements based on specific conditions. it returns a sequence from those 
+    elements of iterable for which function returns True. 
+    :param callback: function with specific filter rules that
+    returns True only for those elements that have to be in filtered result 
+    :param sequence: list, set, tuple with allowed int, float, str values
+    :return: object with int, float, str values
+    """
```
Comment (bohdandovbsyh) [inline HW_7/HW_7.2_Custom_filter_function.py:16]: it is better to use annotations than doc strings.

File: HW_7/HW_7.6_Counting_rotation_function.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    right_counter = 0
+    for each in arrows:
+        if each == '^':
+            up_counter += 1
+        elif each == 'v':
+            down_counter += 1
+        elif each == '<':
+            left_counter += 1
+        elif each == '>':
+            right_counter += 1
```
Comment (bohdandovbsyh) [inline HW_7/HW_7.6_Counting_rotation_function.py]: line from 25 to 37 can be replaced with only 4 lines of code. just remember that in Python most of things are already implemented for you))

Comment (albina-pasichnyk) [reply HW_7/HW_7.6_Counting_rotation_function.py]: Implemented using dictionary.

File: HW_7/HW_7.2_Custom_filter_function.py (hunk 2 above)
Comment (albina-pasichnyk) [reply HW_7/HW_7.2_Custom_filter_function.py:16]: Added annotations in addition to doc string (want to keep it with more explicit explanation).

File: HW_7/HW_7.1_Custom_print_function.py (hunk 1 above)
Comment (albina-pasichnyk) [reply HW_7/HW_7.1_Custom_print_function.py:32]: Thanks

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**